tests/tests_unit/tests_common/test_data_source_local.py

```diff
@@ -103,3 +104,18 @@ def test_attached_messages(demo_data_source: HTTPDataSource):
     )
 
     assert events.filter(lambda event: event.get("attachedMessageIds")).len
+
+
+@pytest.mark.skip(reason="data_source should be changed to mock")
+def test_invalid_optional_timestamp(data_source: HTTPDataSource):
+    with pytest.raises(HTTPError):
+        events = data_source.command(
+            http.GetEventsByBookByScopes(
+                book_id="demo_book_1",
+                scopes=["th2-scope"],
+                start_timestamp=datetime(
+                    year=2022, month=6, day=30, hour=14, minute=0, second=0, microsecond=0
+                ),
+            )
+        )
+        assert events.len
```

Comment:
1. You don't need `data_source` to test it.
2. This test won't find a bug because it has skip status. 

We will create a mock class in the future if we need it.

Reply:
I was trying to raise error where it says if end_timestamp isnt passed either end_timestart or result_limit should be passed.

Tests updated